service/src/loader.rs

```diff
@@ -66,12 +75,42 @@ impl Loader {
         let mut service = self.service;
         let logger = Box::new(Logger::new(tx, deployment_id));
 
-        service.build(factory, logger).await?;
+        fn map_any_to_panic_string(a: &dyn Any) -> String {
+            a.downcast_ref::<&str>()
+                .map(|x| x.to_string())
+                .unwrap_or_else(|| "<no panic message>".to_string())
+        }
```

Comment:
Maybe we should not go with nested functions, but rather put this at the bottom of this file?